File: arcade/intro/56-digitsProduct/digitsProduct.php
function digitsProduct($product) {
    if ($product == 0) {
        return 10;
    }
    if ($product == 1) {
        return 1;
    }
    $r = '';
    for ($i = 9; $i > 1; $i--) {
        if ($product % $i == 0) {
            $product = $product / $i;
            $r = "$i$r;
            $i=10;
            if ($product == 1) {
                return (int)$r;
            }
        }
    }
    return -1;
}

File: arcade/intro/56-digitsProduct/digitsProduct.rb
def digitsProduct(product)
    return 10 if product == 0
    return 1  if product == 1
    p = product
    result = ""
    for i in (9).downto(2)
        while p % i == 0    
            p /= i 
            result = String(i) + result
            puts i
        end
     end   
    p == 1 ? Integer(result) : -1
end